src/bundle.rs

```diff
@@ -50,19 +135,112 @@ pub trait Authorization {
 /// A bundle of actions to be applied to the ledger.
 #[derive(Debug)]
 pub struct Bundle<T: Authorization> {
+    /// The list of actions that make up this bundle.
     actions: NonEmpty<Action<T::SpendAuth>>,
-    flag: Flags,
+    /// Orchard-specific transaction-level flags for this bundle.
+    flags: Flags,
+    /// The net value moved into or out of the Orchard shielded pool.
+    ///
+    /// This is the sum of Orchard spends minus the sum Orchard outputs.
     value_balance: ValueSum,
+    /// The root of the Orchard commitment tree that this bundle commits to.
     anchor: Anchor,
+    /// The authorization for this bundle.
     authorization: T,
 }
 
 impl<T: Authorization> Bundle<T> {
+    /// Constructs a `Bundle` from its constituent parts.
+    pub fn from_parts(
+        actions: NonEmpty<Action<T::SpendAuth>>,
+        flags: Flags,
+        value_balance: ValueSum,
+        anchor: Anchor,
+        authorization: T,
+    ) -> Self {
+        Bundle {
+            actions,
+            flags,
+            value_balance,
+            anchor,
+            authorization,
+        }
+    }
+
+    /// Returns the list of actions that make up this bundle.
+    pub fn actions(&self) -> &NonEmpty<Action<T::SpendAuth>> {
+        &self.actions
+    }
+
+    /// Returns the Orchard-specific transaction-level flags for this bundle.
+    pub fn flags(&self) -> &Flags {
+        &self.flags
+    }
+
+    /// Returns the net value moved into or out of the Orchard shielded pool.
+    ///
+    /// This is the sum of Orchard spends minus the sum Orchard outputs.
+    pub fn value_balance(&self) -> &ValueSum {
+        &self.value_balance
+    }
+
+    /// Returns the root of the Orchard commitment tree that this bundle commits to.
+    pub fn anchor(&self) -> &Anchor {
+        &self.anchor
+    }
+
+    /// Returns the authorization for this bundle.
+    ///
+    /// In the case of a `Bundle<Authorized>`, this is the proof and binding signature.
+    pub fn authorization(&self) -> &T {
+        &self.authorization
+    }
+
     /// Computes a commitment to the effects of this bundle, suitable for inclusion within
     /// a transaction ID.
     pub fn commitment(&self) -> BundleCommitment {
         todo!()
     }
+
+    /// Transitions this bundle from one authorization state to another.
+    pub fn map<U: Authorization>(
```

Comment:
```suggestion
    pub fn authorize<U: Authorization>(
```

Reply:
The rationale for this is that while this is `map`-adjacent, it's not `map` - I think it would be good to have the convention that `map` exclusively means `(a -> b) -> f a -> f b`.

Reply:
I'm applying this change so that I can base additional work on it; if there's disagreement we can revisit later.

Reply:
I disagree because this method does not fully authorize the bundle except in one specific case (where `U` is `Authorized`).

Reply:
Opened #71.